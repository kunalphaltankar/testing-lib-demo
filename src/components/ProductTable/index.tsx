import { IProduct } from "../../models";
import ProductCategoryRow from "../ProductCategoryRow";
import ProductRow from "../ProductRow";

function ProductTable({
  filterText,
  inStockOnly,
  products,
}: {
  filterText: string;
  inStockOnly: boolean;
  products: IProduct[];
}) {
  const rows: React.ReactElement[] = [];
  let lastCategory = '';

  products.forEach((product) => {
    if (product.name.indexOf(filterText) === -1) {
      return;
    }

    if (inStockOnly && !product.stocked) {
      return;
    }

    if (product.category !== lastCategory) {
      rows.push(
        <ProductCategoryRow
          category={product.category}
          key={product.category}
        />,
      );
    }

    rows.push(<ProductRow product={product} key={product.name} />);
    lastCategory = product.category;
  });

  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Price</th>
        </tr>
      </thead>
      <tbody>{rows}</tbody>
    </table>
  );
}

export default ProductTable;