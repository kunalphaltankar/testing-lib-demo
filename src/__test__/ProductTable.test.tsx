import { render } from '@testing-library/react';
import ProductTable from '../components/ProductTable';
import PRODUCTS from '../models/stub/PRODUCTS';

describe('ProductTable', () => {
  it('should render with products', () => {
    const { container } = render(
      <ProductTable products={PRODUCTS} />,
    );
    expect(container).toMatchSnapshot();
  });
  
  it.todo('should show only stocked products');
  it.todo('should filter all products');
  it.todo('should filter only stocked products');
});
