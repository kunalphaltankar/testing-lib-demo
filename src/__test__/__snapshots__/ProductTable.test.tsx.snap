// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable should render with products 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colspan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td>
          Baseball
        </td>
        <td>
          $9.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style="color: red;"
          >
            Basketball
          </span>
        </td>
        <td>
          $29.99
        </td>
      </tr>
      <tr>
        <th
          colspan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          iPod Touch
        </td>
        <td>
          $99.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style="color: red;"
          >
            iPhone 5
          </span>
        </td>
        <td>
          $399.99
        </td>
      </tr>
      <tr>
        <td>
          Nexus 7
        </td>
        <td>
          $199.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
